### PR TITLE
use OpenDefaultApp instead of ActiveRoute.Directions (#1443) (#1864)

### LINKED ISSUE #1443: [POI] Replace ActiveRoute.Directions event with OpenDefaultApp that will pass geo URI for client to consume
POI send event needs to launch the default navigation application with deep link to navigation location.

**Point of Interest Skill**
* When a user selects a route provided by the VA, it will send an event activity with necessary data required for default nav app.

Currently it sends an event `ActiveRoute.Directions` with a JSON payload of Destination and Route Directions. 

- [x] This should be replaced with a new Event:
Name: `OpenDefaultApp`
Value: `geo:latitude,longitude` (coordinates of the destination)

This simplifies integration as it is up to the client to parse how to navigate after receiving instructions from the skill.

**Android App**
- [x] Receive the `OpenDefaultApp` event and consume the parameters required to open default maps app.

### LINKED ISSUE #1864: [POI Skill] Navigation/Routing can break the user state size limit of 2 MB in CosmosDB
#### What project is affected?
POI Skill

#### What language is this in?
C# (TS may be affected too)

#### What happens?
When requesting the navigation feature of POI skill for a long route, the user/conversation state store comes to an Overflow because of the item Limit of 2MB in Cosmos DB. 

#### What are the steps to reproduce this issue?
When requesting the navigation feature of POI skill for a long route, the user/conversation state store comes to an Overflow because of the item Limit of 2MB in Cosmos DB. 

#### What were you expecting to happen?
Regardless of the route I pick no error in the backend.

#### Can you share any logs, error output, etc.?
User State JSON file screenshot:
image

#### Any screenshots or additional context?
There are a lot of Route Direction Points stored in the conversation/user state file so that it exceeds 2MB in total.

### PR DESCRIPTION
**Pull request by xieofxie**
### Purpose
*What is the context of this pull request? Why is it being done?*

Close #1443 and close #1864 

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

* Use OpenDefaultApp instead of ActiveRoute.Directions by CreateOpenDefaultAppReply
* Save RouteDirections.Summary instead of RouteDirections.Route in PointOfInterestSkillState
* Provide maximum 2 alternatives in routes

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

Need test in android.

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

#### Bots
- [ ] I have validated that new and updated responses use appropriate [Speak](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-text-to-speech?view=azure-bot-service-3.0) and [InputHint](https://docs.microsoft.com/en-us/azure/bot-service/dotnet/bot-builder-dotnet-add-input-hints?view=azure-bot-service-3.0) properties to ensure a high-quality speech-first experience
- [ ] I have replicated language model changes across the English, French, Italian, German, Spanish, and Chinese `.lu` files and validated that deployment is successful

#### Deployment Scripts
- [ ] I have replicated my changes in the **Virtual Assistant Template** and **Sample** projects
- [ ] I have replicated my changes in the **Skill Template** and **Sample** projects

### CODE REVIEW

**Review by darrenj — APPROVED** [review]

**Review by lauren-mills — APPROVED** [review]

File: skills/src/csharp/pointofinterestskill/pointofinterestskill/Dialogs/MainDialog.cs [hunk 1]
```diff
@@ -228,12 +228,7 @@ protected override async Task OnEventAsync(DialogContext dc, CancellationToken c
                         var replyMessage = _responseManager.GetResponse(RouteResponses.SendingRouteDetails);
                         await dc.Context.SendActivityAsync(replyMessage);
 
-                        // Send event with active route data
-                        var replyEvent = dc.Context.Activity.CreateReply();
-                        replyEvent.Type = ActivityTypes.Event;
-                        replyEvent.Name = "ActiveRoute.Directions";
-                        replyEvent.Value = state.ActiveRoute.Legs;
-                        await dc.Context.SendActivityAsync(replyEvent);
+                        await dc.Context.SendActivityAsync(PointOfInterestDialogBase.CreateOpenDefaultAppReply(dc.Context.Activity, state.Destination));
```
Comment (lzc850612) [inline skills/src/csharp/pointofinterestskill/pointofinterestskill/Dialogs/MainDialog.cs:231]: if you remove the ActiveRoute.Directions event response, you need to update this doc: https://github.com/microsoft/botframework-solutions/blob/master/docs/reference/skills/pointofinterest.md

**Review by lzc850612 — CHANGES_REQUESTED** [review]

File: skills/src/csharp/pointofinterestskill/pointofinterestskill/Dialogs/MainDialog.cs (hunk 1 above)
Comment (xieofxie) [reply skills/src/csharp/pointofinterestskill/pointofinterestskill/Dialogs/MainDialog.cs:231]: OK, thanks~

**Comment by lzc850612** [comment]
@ryanlengel can you take a look at this PR specifically the doc change since we changed the event?

File: docs/reference/skills/pointofinterest.md [hunk 2]
```diff
@@ -61,39 +61,17 @@ LUIS models for the Skill are provided in .LU file format as part of the Skill.
 |KEYWORD| Simple entity matching point of interest keywords and categories |
 |ROUTE_TYPE| Phrase list entity mapping route descriptors to `eco`,`fastest`,`shortest`,`thrilling`|
 |number| Prebuilt entity|
+|geographyV2| Prebuilt entity|
 
 ## Event Responses
 
-The Point of Interest Skill surfaces a users request to navigate to a new destination through an event returned to the client. The event is called `ActiveRoute.Directions" has contains a series of Points for the Route along with a summary of the route information. A simplified example is shown below
+The Point of Interest Skill surfaces a users request to navigate to a new destination through an event returned to the client. The event is called `OpenDefaultApp" which contains the coordinates of the destination. The format is shown below
```
Comment (ryanisgrig) [inline docs/reference/skills/pointofinterest.md]: Can you replace this section with:

The Point of Interest Skill surfaces a users request to navigate to a new destination through an event returned to the client. The event is called `OpenDefaultApp` which contains the coordinates of the destination. The format is shown below:

```json
{
  "name": "OpenDefaultApp",
  "type": "event",
  "value": "geo:latitude,longitude"
}
```

This event can be customized to return whatever route data is required for your client's needs.

**Review by ryanisgrig — CHANGES_REQUESTED** [review]

File: docs/reference/skills/pointofinterest.md (hunk 2 above)
Comment (ryanisgrig) [reply docs/reference/skills/pointofinterest.md]: I would do it myself but I can't make edits on a fork

Comment (xieofxie) [reply docs/reference/skills/pointofinterest.md]: OK. I will use branch next time..

**Review by ryanisgrig — APPROVED** [review]

**Review by lzc850612 — APPROVED** [review]

**Pull request merged**